src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -91,6 +91,28 @@ private CSharpCompilation CreateNullableCompilation(CSharpTestSource source, IEn
             return CreateCompilation(source, options: WithNonNullTypesTrue(), references: references);
         }
 
+        [Fact, WorkItem(37362, "https://github.com/dotnet/roslyn/issues/37362")]
+        public void DefaultLiteralInConditional()
+        {
+            var comp = CreateNullableCompilation(@"
+class C
+{
+    public void M<T>(bool condition, T t)
+    {
+        t = default; // 1
+        _ = condition ? t : default; // 2
+        _ = condition ? default : t; // 3
```

Comment:
Is the warning is missing here for the same reason as in the linked issue? Just making sure the target type provided by the return statement in the issue's repro doesn't meaningfully affect the way we analyze the conditional.

Reply:
Conditional expressions themselves aren't target typed, the `default` in the conditional should be getting its type from the `t` in the other branch of the conditional. So the target of the assignment (or the return) should not be relevant.

---
In reply to: [314092152](https://github.com/dotnet/roslyn/pull/37596#discussion_r314092152) [](ancestors = 314092152)